tests/publiccloud/register_system.pm

```diff
@@ -27,16 +27,8 @@ sub run {
 
     select_host_console();    # select console on the host, not the PC instance
 
-    my $max_retries = 3;
-    for (1 .. $max_retries) {
-        eval {
-            $args->{my_instance}->run_ssh_command(cmd => "sudo SUSEConnect -r " . get_required_var('SCC_REGCODE'), timeout => 420) unless (get_var('FLAVOR') =~ 'On-Demand');
-        };
-        last unless ($@);
-        diag "SUSEConnect failed: $@";
-        diag "Maybe the SCC or network is busy. Retry: $_ of $max_retries";
-    }
-    die "SCC registration on publiccloud failed (with retries)" if $@;
+    # note: ssh_script_retry dies on failure
+    $args->{my_instance}->ssh_script_retry("sudo SUSEConnect -r " . get_required_var('SCC_REGCODE'), timeout => 420, retry => 3) unless (get_var('FLAVOR') =~ 'On-Demand');
```

Comment:
```suggestion
    $args->{my_instance}->ssh_script_retry("sudo SUSEConnect -r " . get_required_var('SCC_REGCODE'), timeout => 420, retry => 3) if (get_var('FLAVOR') =~ 'BYOS');
```
I think On-Demand was renamed and we now just marking BYOS, or @pdostal @jlausuch 
also see https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11206/files#diff-445d51faa13570063c2cc4756731d747292e18f6b3288df8b544c30f1ac2a5f7L31

Reply:
It's indeed better for the test coverage to do the registration only on systems, that are required to do so. I've incorporated your suggestion